src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -28,11 +30,26 @@ public static IResourceBuilder<RedisResource> AddRedis(this IDistributedApplicat
     {
         ArgumentNullException.ThrowIfNull(builder);
 
-        var redis = new RedisResource(name);
-        return builder.AddResource(redis)
-                      .WithEndpoint(port: port, targetPort: 6379, name: RedisResource.PrimaryEndpointName)
-                      .WithImage(RedisContainerImageTags.Image, RedisContainerImageTags.Tag)
-                      .WithImageRegistry(RedisContainerImageTags.Registry);
+        var resource = new RedisResource(name);
+        var resourceBuilder = builder.AddResource(resource)
+            .WithEndpoint(port: port, targetPort: 6379, name: RedisResource.PrimaryEndpointName)
+            .WithImage(RedisContainerImageTags.Image, RedisContainerImageTags.Tag)
+            .WithImageRegistry(RedisContainerImageTags.Registry)
+            .WithAnnotation(new HealthCheckAnnotation($"StackExchange.Redis_{name}"));
+
+        builder.Eventing.Subscribe<ResourceCreatedEvent>(resource, async (e, ct) =>
+        {
+            var connectionString = await resource.GetConnectionStringAsync(ct).ConfigureAwait(false);
+            builder.Configuration[$"ConnectionStrings:{name}"] = connectionString;
+        });
+
+        builder.AddKeyedRedisClient(name, configureOptions: o =>
```

Comment:
I think this is kind of heavy-weight, isn't it? We just want a HealthCheck registered, not the full client + telemetry, etc.

Why not just add a HealthCheck?

Also, should this be "opt-in" via a new API? So we aren't always doing this.

Reply:
I think we could try to make it lighter but we'd probably end up with pretty similar code to what we have in the Aspire integration packages - so we may as well just reuse them.

The main wrinkle is that we generally assume that the connection string is available at the point that we register the dependencies with DI - an assumption that does not hold in this scenario because we don't learn the connection string until after the container is built and endpoints are allocated.

So far I've managed to work around it but I'm not sure how long my luck will hold :)